ui/app/routes/vault/cluster/secrets/backend/secret-edit.js

```diff
@@ -81,17 +81,27 @@ export default Route.extend({
       const mode = this.routeName.split('.').pop();
       // for kv v2, redirect users from the old url to the new engine url (1.15.0 +)
       if (secretEngine.type === 'kv' && secretEngine.version === 2) {
```

Comment:
Since we're cleaning up the redirect logic, I believe we should also redirect for `secretEngine.type === 'generic' && secretEngine.version === 2`

Reply:
I'm hesitant to change or add anything to this logic flow, but happy to do this as a follow on! The only improvement/change I made by reorganizing the logic is to navigate to `edit` or `index` accordingly instead of arbitrarily go to `details`. I think during the original implementation we weren't aware we had the route info available to discern between a user `either viewing or editing secret so navigate to the details route`